packit/base_git.py

```diff
@@ -0,0 +1,95 @@
+import git
+import logging
+from typing import Optional
+
+from packit.config import Config, PackageConfig
+from packit.exceptions import PackitException
+from packit.local_project import LocalProject
+
+logger = logging.getLogger(__name__)
+
+
+class BaseGitClass:
+    def __init__(self, config: Config, package_config: PackageConfig):
+        self.config = config
+        self.package_config = package_config
+        self.package_name: Optional[str] = self.package_config.downstream_package_name
+        self.dist_git_url: Optional[str] = self.package_config.downstream_project_url
```

Comment:
If the properties are not used in this class, I will leave them in the subclass. (It looks like they're not used in the `Upstream`/`SourceGit`.)